src/main/java/com/google/devtools/build/lib/runtime/UiStateTracker.java

```diff
@@ -193,6 +194,13 @@ String formatNames(int bitmap) {
      */
     boolean scanning;
 
+    /**
+     * Bitmap of strategies that are checking the cache of this action.
+     *
+     * <p>If non-zero, implies that {@link #scanning} is false.
+     */
+    int cachingStrategiesBitmap = 0;
```

Comment:
Can we use an https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/EnumSet.html here instead of manually handling a bitmap?

I see that `schedulingStrategiesBitmap` and `runningStrategiesBitmap ` were already there before, but maybe it's an opportunity to make the code more readable 😊

Reply:
As discussed, since the bit is generated dynamically, we will go with how the existing code does.